app/models/media_object.rb

```diff
@@ -35,6 +35,7 @@ class MediaObject < ActiveFedora::Base
 
   after_create :after_create
   before_save :normalize_desc_metadata!
+  before_save :update_dependent_properties!
```

Comment:
Maybe making this after_save would solve the test failure you're getting.  Before your PR this was happening after the yield in an around_save.  Ah, but that would cause double-saving so maybe we just need to make the mods datastream autocreate like here: https://github.com/avalonmediasystem/avalon/blob/develop/app/models/admin/collection.rb#L35